src/actions/mod.rs

```diff
@@ -98,13 +98,15 @@ impl ActionHandler {
                                                   project_path: &Path)
             -> Vec<NotificationMessage<PublishRustDiagnosticsParams>>
         {
+            let cwd = ::std::env::current_dir().unwrap();
+
             build_results
             .iter()
             .map(|(path, diagnostics)| {
                 let method = "textDocument/publishDiagnostics".to_string();
 
                 let params = PublishRustDiagnosticsParams {
-                    uri: Url::from_file_path(project_path.join(path)).unwrap(),
+                    uri: Url::from_file_path(cwd.join(path)).unwrap(),
```

Comment:
Ahhh, okay, that was the piece that wasn't clicking.  Now it makes sense.

There's someone on twitter testing this patch out now: https://twitter.com/tyroverby/status/854804696879452160

I suspect we'll want to take this fix and then start planning how to get out of using cwd.